client/src/controllers/BulkController.ts

```diff
@@ -0,0 +1,72 @@
+import { ChangeEvent } from 'react';
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Controller that adds the ability to toggle some or all the elements, ignoring the disabled ones
+ * Additionaly, it checks if all non-disabled items are selected or not. If they are, then select all
+ *
+ * @example
+ * <div data-controller="w-bulk">
+ *   <input type="checkbox" data-w-bulk-target="all">
+ *
+ *   <div>
+ *     <input type="checkbox" data-w-bulk-target="item" disabled>
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *   </div>
+ *
+ *   <button data-action="w-bulk#clear">Clear</button>
```

Comment:
I like this idea, a button to 'clear' the checkboxes, however we would probably do this like so.

```suggestion
 *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
  *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
```

This is a really nice feature of Stimulus where you can 'add' some other param to the event via data attributes, see other comment below for the small change needed to `toggleAll` to read this value.

Also, let's write a unit test for this.